mypy/build.py

```diff
@@ -861,19 +865,75 @@ def is_file(path: str) -> bool:
     return res
 
 
-def find_module(id: str, lib_path_arg: Iterable[str]) -> Optional[str]:
+USER_SITE_PACKAGES = \
+    'from __future__ import print_function; import site; print(site.getusersitepackages());' \
+    'print(*site.getsitepackages(), sep="\\n")'
+VIRTUALENV_SITE_PACKAGES = \
+    'from distutils.sysconfig import get_python_lib; print(get_python_lib())'
+
+
+def call_python(python: str, command: str) -> str:
+    return subprocess.check_output([python, '-c', command], stderr=subprocess.PIPE).decode('UTF-8')
+
+
+def get_package_dirs(python_executable: str) -> List[str]:
+    """Find package directories for given python
+
+     This defaults to the Python running mypy.
+     """
+    if python_executable in package_dirs_cache:
+        return package_dirs_cache[python_executable]
+    package_dirs = []  # type: List[str]
+    if python_executable == sys.executable:
+        # Use running Python's package dirs
+        try:
+            user_dir = site.getusersitepackages()
+            package_dirs = site.getsitepackages() + [user_dir]
+        except AttributeError:
+            package_dirs = [get_python_lib()]
+    else:
+        # Use subprocess to get the package directory of given Python
+        # executable
+        try:
+            output = call_python(python_executable, USER_SITE_PACKAGES)
+        except subprocess.CalledProcessError:
+            output = ''
+        for line in output.splitlines():
+            if os.path.isdir(line):
+                package_dirs.append(line)
+        # if no paths are found, we fall back on sysconfig, the python is likely in a
```

Comment:
I don't see where the `if no paths are found` is coming into play here - it looks like you just do it always